src/pages/SearchPage.js

```diff
@@ -101,48 +106,106 @@ class SearchPage extends Component {
         }];
 
         if (this.state.userToInvite) {
-            sections.push(({
+            sections.push({
                 undefined,
                 data: [this.state.userToInvite],
                 shouldShow: true,
                 indexOffset: 0,
-            }));
+            });
         }
 
         return sections;
     }
 
     /**
-     * Reset the search value and redirect to the selected report
-     *
-     * @param {Object} option
-     */
+   * Reset the search value and redirect to the selected report
+   * @param {Object} option
+   */
     selectReport(option) {
         if (!option) {
             return;
         }
 
         if (option.reportID) {
+            this.setState(
+                {
+                    searchValue: '',
+                },
+                () => {
+                    Navigation.navigate(ROUTES.getReportRoute(option.reportID));
+                },
+            );
+        } else {
+            fetchOrCreateChatReport([this.props.session.email, option.login]);
+        }
+    }
+
+    validateInput(searchValue) {
+        if (!searchValue) {
+            return;
+        }
+
+        let modifiedSearchValue = searchValue;
+        const headerMessage = getHeaderMessage(
+            this.state.personalDetails.length !== 0,
+            false,
+            searchValue,
+        );
+
+        if (/^\d+$/.test(searchValue) || /^[0-9]*$/.test(searchValue)) {
```

Comment:
I tried using `Str.isValidPhone` but it is not working properly. Also I am using `||` so that if any of the regex matches then it skips the checks for second one. Combining them is possible but will increase the time complexity of the program.